### PR TITLE
Use ruff

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Use `ruff` for linting and formatting,
- [x] remove lint GitHub workflow as it is redundant with pre-commit,
- [n/a] docstring updated (if appropriate),
- [n/a] update user guide (if appropriate),
- [n/a] added tests,
- [x] add a changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/exspy/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting of the changelog entry (and eventual user guide changes) in the `docs/readthedocs.org:exspy` build of this PR (link in github checks)
- [x] ready for review.

### CODE REVIEW

File: exspy/tests/components/test_double_power_law.py [hunk 1]
```diff
@@ -46,11 +46,11 @@ def test_function():
 
 class TestDoublePowerLaw:
     def setup_method(self, method):
+        pytest.importorskip("exspy")
```
Comment (jlaehne) [inline exspy/tests/components/test_double_power_law.py:49]: In what situation would `exspy` not be available for its own tests?

File: upcoming_changes/README.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
+This directory contains "news fragments" which are short files that contain a small **ReST**-formatted
+text that will be added to the next ``CHANGELOG``.
+
+The ``CHANGELOG`` will be read by **users**, so this description should be aimed to hyperspy users
```
Comment (jlaehne) [inline upcoming_changes/README.rst:4]:
```suggestion
The ``CHANGELOG`` will be read by **users**, so this description should be aimed to eXSpy users
```

Comment (ericpre) [reply upcoming_changes/README.rst:4]: Only noticed that you already added this file in https://github.com/hyperspy/exspy/pull/22. Will rebase after #22 is merged.

File: exspy/tests/components/test_double_power_law.py (hunk 1 above)
Comment (ericpre) [reply exspy/tests/components/test_double_power_law.py]: Indeed, it doesn't make sense!

**Review by jlaehne — APPROVED** [review]

**Pull request merged**